### DISCUSSION TITLE
'ZBlazorDiagrams' was undefined

### DISCUSSION THREAD

**Discussion by PiggyChu620** [Blazor-Diagrams/Blazor.Diagrams#442] [Q&A]
Hello guys,
I tried to clone the whole repo and integrates it directly into my project instead of installing it from NuGet, so I have full control over it to do something like integreting pending pull requests before the author merging it.
Anyway, I am on the tight schedule and therefore need it to be done asap, so I have no choice but to integrate the package manually instead of waiting and hoping the author merges it.
Don't get me wrong, I appreciate the author, from the bottom of my heart, releasing such a great tool so that I could do something completely impossible on my own, I just can not wait passively.
Anyway, again, I managed to fix all the referecing issues and get it running, but when I open the diagram page, an unhandled exception occured, and when I check the output, it says "'ZBlazorDiagrams' was undefined", but I did change the references in _Host.cshtml to its new location, so why is it still failing?
Could somebody please be so kind telling me what have I missed?
Much appreciated!

**Comment by Basyras** [comment]
the package JS is not reference correctly.
in index.html file you need to reference the packge js module. When you reference the package via nuget its done like this:
```
index.html
<script src="_content/Z.Blazor.Diagrams/script.min.js"></script>
```

It depends how did you integrate the source code, maybe the script path needs to be "local" so try:
```
index.html
<script src="_content/script.min.js"></script>
```

you might also try blazor's javascript [initiliazers](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0)
